x-pack/plugins/infra/server/lib/snapshot/create_timerange_with_interval.ts

```diff
@@ -12,30 +12,48 @@ import { SnapshotModel, SnapshotModelMetricAggRT } from '../../../common/invento
 import { getDatasetForField } from '../../routes/metrics_explorer/lib/get_dataset_for_field';
 import { InfraTimerangeInput } from '../../../common/http_api/snapshot_api';
 import { ESSearchClient } from '.';
+import { getIntervalInSeconds } from '../../utils/get_interval_in_seconds';
 
-export const createTimeRangeWithInterval = async (
-  client: ESSearchClient,
-  options: InfraSnapshotRequestOptions
-): Promise<InfraTimerangeInput> => {
+const createInterval = async (client: ESSearchClient, options: InfraSnapshotRequestOptions) => {
+  const { timerange } = options;
+  if (timerange.forceInterval && timerange.interval) {
+    return getIntervalInSeconds(timerange.interval);
+  }
   const aggregations = getMetricsAggregations(options);
   const modules = await aggregationsToModules(client, aggregations, options);
-  const interval = Math.max(
+  return Math.max(
     (await calculateMetricInterval(
       client,
       {
         indexPattern: options.sourceConfiguration.metricAlias,
         timestampField: options.sourceConfiguration.fields.timestamp,
-        timerange: { from: options.timerange.from, to: options.timerange.to },
+        timerange: { from: timerange.from, to: timerange.to },
       },
       modules,
       options.nodeType
     )) || 60,
     60
   );
+};
+
+export const createTimeRangeWithInterval = async (
+  client: ESSearchClient,
+  options: InfraSnapshotRequestOptions
+): Promise<InfraTimerangeInput> => {
+  const { timerange } = options;
+  const calculatedInterval = await createInterval(client, options);
+  if (timerange.ignoreLookback) {
+    return {
+      interval: `${calculatedInterval}s`,
+      from: timerange.from,
+      to: timerange.to,
+    };
+  }
+  const lookbackSize = timerange.lookbackSize || 5;
   return {
-    interval: `${interval}s`,
-    from: options.timerange.to - interval * 5000, // We need at least 5 buckets worth of data
-    to: options.timerange.to,
+    interval: `${calculatedInterval}s`,
+    from: timerange.to - calculatedInterval * lookbackSize * 1000, // We need at least 5 buckets worth of data
```

Comment:
Maybe we should set a minimum `lookbackSize` of 5? I'm not sure if we'll get into a situation where this API might get sent a number lower than 5, but I don't think we want to leave it up to developer discipline to just never send a number that wouldn't work.

Reply:
Good idea 👍